src/mg_decoration.cpp

```diff
@@ -304,13 +304,16 @@ size_t DecoSimple::generate(MMVManip *vm, PcgRandom *pr, v3s16 p)
 	s16 height = (deco_height_max > 0) ?
 		pr->range(deco_height, deco_height_max) : deco_height;
 
+	bool force_placement = (flags & DECO_FORCE_PLACEMENT);
```

Comment:
But then, almost all simple decorations are a single node so using the flag directly would be more efficient overall.